src/microbe_stage/organelle_unlocks/StatisticBasedUnlockCondition.cs

```diff
@@ -0,0 +1,188 @@
+﻿namespace UnlockConstraints
+{
+    using Newtonsoft.Json;
+
+    /// <summary>
+    ///   An unlock condition that relies on a certain <see cref="IStatistic"/> to track progress
+    /// </summary>
+    public abstract class StatisticBasedUnlockCondition : IUnlockCondition
+    {
+        [JsonIgnore]
+        public abstract StatsTrackerEvent RelevantEvent { get; }
+
+        [JsonIgnore]
+        public IStatistic? RelevantStatistic { get; set; }
+
+        public abstract bool Satisfied();
+
+        public abstract void GenerateTooltip(LocalizedStringBuilder builder);
+
+        public virtual void Check(string name)
+        {
+        }
+
+        public virtual void Resolve(SimulationParameters parameters)
+        {
+        }
+    }
+
+    /// <summary>
+    ///   The number of microbes engulfed by the player is above the specified amount.
+    /// </summary>
+    public class EngulfedMicrobesAbove : StatisticBasedUnlockCondition
+    {
+        public override StatsTrackerEvent RelevantEvent => StatsTrackerEvent.PlayerEngulfedOther;
+
+        [JsonProperty]
+        public int Required { get; set; }
+
+        [JsonIgnore]
+        public SimpleStatistic? Engulfed => (SimpleStatistic?)RelevantStatistic;
```

Comment:
I think making the base class generic would be better than needing to rely on casting in the derived class here. So this would inherit `StatisticBasedUnlockCondition<SimpleStatistic>`.